File: applications-fn/Application.cs
using System.Collections.Generic;

namespace applications_fn
{
    public class Application 
    {
        public int Id { get; set; }
        public string CandidateName { get; set; }
        public Availability Availability { get; set; }
        public List<Question> Questions { get; set; }
    }    
}

File: web/src/app/types/application.ts
import { Availability } from './availability';
import { Question  } from './question';

export class Application {
    id: number;
    applied: Date;
    name: string;
    position: string;
    questions: Question[];
    experience: number;
    availability: Availability;
    // application layer properties
    isFavorited: boolean;
    isBookmarked: boolean;
    isMarkedForLater: boolean;
}
